packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -73,14 +73,11 @@ def token_ids(self) -> Dict[int, int]:
     def create_token_ids(self, token_type: int, nb_tokens: int) -> List[int]:
         """Populate the token_ids dictionary."""
         assert self.token_ids == {}, "Item ids already created."
-        lowest_valid_integer = 0
-        token_id = Helpers().generate_id(token_type, lowest_valid_integer)
+        lowest_valid_integer = 1
         token_id_list = []
-        for _i in range(nb_tokens):
-            while self.instance.functions.is_token_id_exists(token_id).call():
-                # id already taken
-                lowest_valid_integer += 1
-                token_id = Helpers().generate_id(token_type, lowest_valid_integer)
```

Comment:
There is no point on checking if it is created on the contract side. Here we just creating the ids locally. Maybe it makes more sense to check this on the create_batch function.

Reply:
For sure it makes sense, you would not want to generate ids which you then cannot use. Please revert.

Reply:
the lowest valid integer must be 1. Then We generate the one token_id. and then we go in the for loop. If the item exists we go in the While loop. But since this is a locally generating_id and not on the contract. This is going to always be false. The lowest_valid_integer will always be 0 and we are going to append ten times the lowest_valid_id in the list and in the dictionary.

Reply:
I see, so you should change the code to:
```
lowest_valid_integer = 1
token_id = Helpers().generate_id(token_type, lowest_valid_integer)
token_id_list = []
for _i in range(nb_tokens):
    while self.instance.functions.is_token_id_exists(token_id).call()
            # token_id already taken
            lowest_valid_integer += 1
            token_id = Helpers().generate_id(token_type, lowest_valid_integer)
    token_id_list.append(token_id)
    self.token_ids[token_id] = token_type
    lowest_valid_integer += 1
    token_id = Helpers().generate_id(token_type, lowest_valid_integer)
```